aws/resource_aws_workspaces_workspace.go

```diff
@@ -366,12 +366,21 @@ func expandWorkspaceProperties(properties []interface{}) *workspaces.WorkspacePr
 
 	p := properties[0].(map[string]interface{})
 
-	return &workspaces.WorkspaceProperties{
-		ComputeTypeName:                     aws.String(p["compute_type_name"].(string)),
-		RootVolumeSizeGib:                   aws.Int64(int64(p["root_volume_size_gib"].(int))),
-		RunningMode:                         aws.String(p["running_mode"].(string)),
-		RunningModeAutoStopTimeoutInMinutes: aws.Int64(int64(p["running_mode_auto_stop_timeout_in_minutes"].(int))),
-		UserVolumeSizeGib:                   aws.Int64(int64(p["user_volume_size_gib"].(int))),
+	if p["running_mode"] == workspaces.RunningModeAlwaysOn {
+		return &workspaces.WorkspaceProperties{
+			ComputeTypeName:   aws.String(p["compute_type_name"].(string)),
+			RootVolumeSizeGib: aws.Int64(int64(p["root_volume_size_gib"].(int))),
+			RunningMode:       aws.String(p["running_mode"].(string)),
+			UserVolumeSizeGib: aws.Int64(int64(p["user_volume_size_gib"].(int))),
+		}
+	} else {
+		return &workspaces.WorkspaceProperties{
+			ComputeTypeName:                     aws.String(p["compute_type_name"].(string)),
+			RootVolumeSizeGib:                   aws.Int64(int64(p["root_volume_size_gib"].(int))),
+			RunningMode:                         aws.String(p["running_mode"].(string)),
+			RunningModeAutoStopTimeoutInMinutes: aws.Int64(int64(p["running_mode_auto_stop_timeout_in_minutes"].(int))),
+			UserVolumeSizeGib:                   aws.Int64(int64(p["user_volume_size_gib"].(int))),
+		}
```

Comment:
Nit: since both conditions set similar attributes in the `WorkspacesProperties` object, we could initialize the object first and then run through the if/else logic e.g.
```suggestion
        workspaceProperties := &workspaces.WorkspaceProperties{
			ComputeTypeName:   aws.String(p["compute_type_name"].(string)),
			RootVolumeSizeGib: aws.Int64(int64(p["root_volume_size_gib"].(int))),
			RunningMode:       aws.String(p["running_mode"].(string)),
			UserVolumeSizeGib: aws.Int64(int64(p["user_volume_size_gib"].(int))),
	}
	if p["running_mode"] == workspaces.RunningModeAlwaysOn {
	    workspaceProperties.RunningModeAutoStopTimeoutInMinutes = aws.Int64(int64(p["running_mode_auto_stop_timeout_in_minutes"].(int)))
        }
```